src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -0,0 +1,1192 @@
+        // No inline assertions for expando members
+        if (expandoFunction) return;
+        const targetNode = findBestFittingNode(nodeWithDiag, span);
+        if (!targetNode || isValueSignatureDeclaration(targetNode) || isValueSignatureDeclaration(targetNode.parent)) return;
+        const isExpressionTarget = isExpression(targetNode);
+        const isShorthandPropertyAssignmentTarget = isShorthandPropertyAssignment(targetNode);
+
+        if (!isShorthandPropertyAssignmentTarget && isDeclaration(targetNode)) {
+            return undefined;
+        }
+        // No inline assertions on binding patterns
+        if (findAncestor(targetNode, isBindingPattern)) {
+            return undefined;
+        }
+
+        // No inline assertions on enum members
+        if (findAncestor(targetNode, isEnumMember)) {
+            return undefined;
+        }
+        // No support for typeof in extends clauses
+        if (isExpressionTarget && findAncestor(targetNode, isHeritageClause)) {
+            return undefined;
+        }
+        // Can't inline type spread elements. Whatever you do isolated declarations will not infer from them
+        if (isSpreadElement(targetNode)) {
+            return undefined;
+        }
+
+        const variableDeclaration = findAncestor(targetNode, isVariableDeclaration);
+        const type = variableDeclaration && typeChecker.getTypeAtLocation(variableDeclaration);
+        // We can't use typeof un an unique symbol. Would result in either
+        // const s = Symbol("") as unique symbol
+        // const s = Symbol("") as typeof s
+        // both of which are not correct
+        if (type && type.flags & TypeFlags.UniqueESSymbol) {
+            return undefined;
+        }
+
+        if (!(isExpressionTarget || isShorthandPropertyAssignmentTarget)) return undefined;
+
+        const { typeNode, mutatedTarget } = inferType(targetNode, type);
+        if (!typeNode || mutatedTarget) return undefined;
+
+        if (isShorthandPropertyAssignmentTarget) {
+            changeTracker.insertNodeAt(
+                sourceFile,
+                targetNode.end,
+                createAsExpression(
+                    getSynthesizedDeepClone(targetNode.name),
+                    typeNode,
+                ),
+                {
+                    prefix: ": ",
+                },
+            );
+        }
+        else if (isExpressionTarget) {
+            changeTracker.replaceNode(
+                sourceFile,
+                targetNode,
+                createSatisfiesAsExpression(
+                    getSynthesizedDeepClone(targetNode),
+                    typeNode,
+                ),
+            );
+        }
+        else {
+            Debug.assertNever(targetNode);
+        }
+        return [Diagnostics.Add_satisfies_and_an_inline_type_assertion_with_0, typeToStringForDiag(typeNode)];
+    }
+
+    function suggestVariableName(node: Node) {

```

Comment:
I'm a little skeptical of spending this many lines generating a name beyond a single level of valid-identifier property assignment when the user will almost certainly replace it after extracting. There's a one-liner best effort in `extractSymbol.ts` that falls back to `getUniqueName("newLocal")` that I think would suffice here. Maybe you could extract that to a common utility, improving it if needed (but only insofar as the improvement results in a high-confidence suggestion).

It would be breat if we could automatically invoke Rename after the codefix like we do in the Extract Symbol refactor, but I don't think our current protocol allows for that.

Reply:
Removed and replaced the usages with the factored-out function, does this look sense to you?